module/support/SoccerSimulator/src/VirtualGoalPost.cpp

```diff
@@ -112,67 +115,113 @@ namespace support {
+
+                goal.measurements.erase(goal.measurements.begin() + 3);
             }
             // goal top visibility check
             if (not(quad.getTopRight()[1] > 0 && quad.getTopRight()[1] < image.dimensions[1] && quad.getTopLeft()[1] > 0
-                    && quad.getTopLeft()[1] < image.dimensions[1] && quad.getTopRight()[0] > 0
-                    && quad.getTopRight()[0] < image.dimensions[0] && quad.getTopLeft()[0] > 0
+                    && quad.getTopLeft()[1] < image.dimensions[1]
+                    && quad.getTopRight()[0] > 0
+                    && quad.getTopRight()[0] < image.dimensions[0]
+                    && quad.getTopLeft()[0] > 0
                     && quad.getTopLeft()[0] < image.dimensions[0])) {
 
-                result.goals.at(0).measurements.erase(result.goals.at(0).measurements.begin() + 2);
+                goal.measurements.erase(goal.measurements.begin() + 2);
             }
+
             // goal sides visibility check
             if (not((
                         // One of the top or the bottom are in the screen coordinates of x
                         (quad.getBottomLeft()[0] > 0 && quad.getBottomLeft()[0] < image.dimensions[0]
-                         && quad.getBottomRight()[0] > 0 && quad.getBottomRight()[0] < image.dimensions[0])
+                         && quad.getBottomRight()[0] > 0
+                         && quad.getBottomRight()[0] < image.dimensions[0])
                         || (quad.getTopLeft()[0] > 0 && quad.getTopLeft()[0] < image.dimensions[0]
-                            && quad.getTopRight()[0] > 0 && quad.getTopRight()[0] < image.dimensions[0]))
+                            && quad.getTopRight()[0] > 0
+                            && quad.getTopRight()[0] < image.dimensions[0]))
                     && (
                            // Check that the bottom is below the top of the screen and the top is below the bottom of
+
                            // the screen
                            (quad.getBottomRight()[1] < image.dimensions[1] && quad.getTopRight()[1] > 0)
                            || (quad.getBottomLeft()[1] < image.dimensions[1] && quad.getTopLeft()[1] > 0)))) {
-                result.goals.at(0).measurements.erase(result.goals.at(0).measurements.begin() + 1);
-                result.goals.at(0).measurements.erase(result.goals.at(0).measurements.begin());
+                goal.measurements.erase(goal.measurements.begin() + 1);
+                goal.measurements.erase(goal.measurements.begin());
             }
-            // if (!result.goals.at(0).measurements.empty()) {
-            //     result.goals.at(0).post.top    = (convert(quad.getTopLeft()) + convert(quad.getTopRight()) * 0.5);
-            //     result.goals.at(0).post.bottom = (convert(quad.getBottomLeft()) + convert(quad.getBottomRight()) *
-            //     0.5);
-            // }
+
+            // if (!goal.measurements.empty()) {
```

Comment:
I commented this out because I couldn't be bothered trying to fix it. Since you have fixed everything else in this file you should uncomment this block and fix it too